shared-libs/environment/src/index.js

```diff
@@ -57,5 +63,48 @@ try {
   process.exit(1);
 }
 
+let deployInfoCache;
+
+const getVersionFromDdoc = (ddoc) => {
+  return semver.valid(ddoc.build_info?.version) ||
+    semver.valid(ddoc.deploy_info?.build) ||
+    ddoc.version ||
+    'unknown';
+};
+
+const getDeployInfo = async () => {
+  if (deployInfoCache) {
+    return deployInfoCache;
+  }
+
+  let ddoc;
+  try {
+    ddoc = await request.get({
```

Comment:
I believe `ddoc` can be a scoped `const` here, and we can return within the try-catch block.